client/my-sites/customer-home/controller.jsx

```diff
@@ -38,14 +38,18 @@ export async function maybeRedirect( context, next ) {
 
 	const siteId = getSelectedSiteId( state );
 	const maybeStalelaunchpadScreenOption = getSiteOptions( state, siteId )?.launchpad_screen;
+	const currentUrl = window.location.href;
 
 	// We need the latest site info to determine if a user should be redirected to Launchpad.
 	// As a result, we can't use locally cached data, which might think that Launchpad is
 	// still enabled when, in reality, the final tasks have been completed and everything is
 	// disabled. Because of this, we refetch site information and limit traffic by scoping down
 	// requests to launchpad enabled sites.
 	// See https://cylonp2.wordpress.com/2022/09/19/question-about-infinite-redirect/#comment-1731
-	if ( maybeStalelaunchpadScreenOption && maybeStalelaunchpadScreenOption === 'full' ) {
+	if (
+		( maybeStalelaunchpadScreenOption && maybeStalelaunchpadScreenOption === 'full' ) ||
+		currentUrl?.includes( 'postOnboarding=true' )
```

Comment:
should we get and read the query args specifically here?

Reply:
side note, if we are only adding the arg from launchpad flows, then we could maybe just use that itself as the indication to redirect back to the launchpad here as opposed to refetching? Although, maybe the refetching is good to do in the first place anyways to make sure calypso store is truly up to date.

Reply:
> Could you elaborate on this?

Sure, its probably not a big deal at all but right now we are just parsing the entire string of the URL to see if "postOnboarding=true" is part of the string. As opposed to getting the specific params out of the URL and checking their values. This might generally be something like [$_GET ](https://www.php.net/manual/en/reserved.variables.get.php) in php or maybe [searchParams](https://developer.mozilla.org/en-US/docs/Web/API/URL/searchParams) in JS (maybe theres another method im not remembering). That said, Its probably not a big deal. We aren't going to see `postOnboarding=true` elsewhere in the URL string so it should work the same way.

> If we don't force a refetch, even if we redirect to Launchpad, the site option data in redux still might be stale if they happen to navigate to my home. I believe if we visit /view or the editors the cache should be invalidated, but I'm not sure

That makes sense 👍